File: src/Leetcode/1_twoSum.js
/*
https://leetcode.com/problems/two-sum/
Given an array of integers and an integer "target" return indices of the two numbers such that
they add up to target.

Assume each array has one solution.
 */

var twoSum = function(nums, target) { // 96 ms solution
    for(let i = 0; i < nums.length; i++){
        let x = target - nums[i];
        for(let j = i+1; j < nums.length; j++){
            if(nums[j] === x){
                return [i,j]
            }
        }
    }
};

console.log(twoSum([3,3],6))

File: src/Python/1_two_sum.py
class Solution(object):
    def twoSum(self, nums, target):
        for x in range(0,len(nums)):
            t = target - nums[x]
            for y in range(x+1, len(nums)):
                if t == nums[y]:
                    return [y, x]

solution = Solution()
nums = [2,7,11,5]
target = 9
result = solution.twoSum(nums,target)
print(result)